### PR TITLE
Update Maven registry config

### PR DESCRIPTION
**Pull request by pkwarren**
Update Maven registry config to support customizing compiler settings,
adding support for multiple runtimes (like the 'lite' runtime), and more
settings for runtime dependencies (classifier and file type).

### CODE REVIEW

File: private/bufpkg/bufplugin/bufplugin_test.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func TestPluginRegistryRoundTrip(t *testing.T) {
 	})
 	assertPluginRegistryRoundTrip(t, &bufpluginconfig.RegistryConfig{
 		NPM: &bufpluginconfig.NPMRegistryConfig{
-			ImportStyle: "module",
+			ImportStyle: "commonjs",
```
Comment (stefanvanburen) [inline private/bufpkg/bufplugin/bufplugin_test.go:60]: wasn't sure if this was changed intentionally?

**Review by stefanvanburen — COMMENTED** [review]
👍, minor nit on the npm test change

File: private/bufpkg/bufplugin/bufplugin_test.go (hunk 1 above)
Comment (pkwarren) [reply private/bufpkg/bufplugin/bufplugin_test.go:60]: Yeah sorry meant to add a comment. Noticed a branch with no test coverage - by changing this it was exercised.

File: proto/buf/alpha/registry/v1alpha1/plugin_curation.proto [hunk 2]
```diff
@@ -96,14 +96,51 @@ message NPMConfig {
...
-    string gav = 1;
+    string group_id = 1;
+    string artifact_id = 2;
+    string version = 3;
+    string classifier = 4;
+    string extension = 5;
+  }
+  // CompilerConfig contains configuration for the Java and/or Kotlin compiler used when compiling the generated code.
+  message CompilerConfig {
+    CompilerJavaConfig java = 1;
```
[the remark this answers is not in the document]
Comment (pkwarren) [reply proto/buf/alpha/registry/v1alpha1/plugin_curation.proto:109]: Thought about this - while unlikely, someone could author a plugin that emits both Java and Kotlin sources. In that case, we'd use something like https://kotlinlang.org/docs/maven.html#compile-kotlin-and-java-sources to compile both.

File: private/bufpkg/bufplugin/bufpluginconfig/bufpluginconfig.go [hunk 3]
```diff
@@ -293,7 +346,51 @@ type ExternalNPMRegistryConfig struct {
...
+	// LanguageVersion is used to provide source compatibility with the specified Kotlin version.
+	LanguageVersion string `json:"language_version" yaml:"language_version"`
+	// Version of the Kotlin compiler to use (required for Kotlin plugins).
+	Version string `json:"version" yaml:"version"`
+}
+
+// ExternalMavenRuntime allows configuring additional runtimes for remote packages.
+// These can specify different dependencies and compiler options than the default runtime.
+// This is used to support a single plugin supporting both full and lite Protobuf runtimes.
+type ExternalMavenRuntime struct {
```
Comment (stefanvanburen) [inline private/bufpkg/bufplugin/bufpluginconfig/bufpluginconfig.go:387]: should we make this `ExternalMavenRuntimeConfig` to be inline with the other structs here?

**Review by stefanvanburen — APPROVED** [review]

File: private/bufpkg/bufplugin/bufpluginconfig/bufpluginconfig.go (hunk 3 above)
Comment (pkwarren) [reply private/bufpkg/bufplugin/bufpluginconfig/bufpluginconfig.go]: Fixed.

**Review by stefanvanburen — APPROVED** [review]

**Pull request merged**